crates/uv-toolchain/src/discovery.rs

```diff
@@ -1268,8 +1275,33 @@ impl FromStr for VersionRequest {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // e.g. `3`, `38`, `312`
+        if let Some(request) = {
```

Comment:
It took me a while to parse this. (With a huge expression on the right hand side of an `if let`.) I'd probably move it up and give it a name?